recipes/libusb-compat/all/conanfile.py

```diff
@@ -0,0 +1,156 @@
+    url = "https://github.com/conan-io/conan-center-index"
+    exports_sources = "patches/**"
+    topics = ("conan", "libusb", "compatibility", "usb")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "enable_logging": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "enable_logging": False,
+    }
+    generators = "pkg_config"
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("libusb-compat-{}".format(self.version), self._source_subfolder)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def requirements(self):
+        self.requires("libusb/1.0.23")
+        if self.settings.compiler == "Visual Studio":
+            self.requires("dirent/1.23.2")
+
+    def build_requirements(self):
+        self.build_requires("libtool/2.4.6")
+        if tools.os_info.is_windows and not os.environ.get("CONAN_BASH_PATH") and \
+                tools.os_info.detect_windows_subsystem() != "msys2":
+            self.build_requires("msys2/20190524")
+
+    def _iterate_lib_paths_win(self, lib):
+        """Return all possible library paths for lib"""
+        for lib_path in self.deps_cpp_info.lib_paths:
+            for prefix in "", "lib":
+                for suffix in "", ".a", ".dll.a", ".lib", ".dll.lib":
+                    fn = os.path.join(lib_path, "{}{}{}".format(prefix, lib, suffix))
+                    if not fn.endswith(".a") and not fn.endswith(".lib"):
+                        continue
+                    yield fn
+
+    @property
+    def _absolute_dep_libs_win(self):
+        absolute_libs = []
+        for lib in self.deps_cpp_info.libs:
+            for fn in self._iterate_lib_paths_win(lib):
+                if not os.path.isfile(fn):
+                    continue
+                absolute_libs.append(fn)
+                break
+        return absolute_libs
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
+        if self.settings.compiler == "Visual Studio":
+            # Use absolute paths of the libraries instead of the library names only.
+            # Otherwise, the configure script will say that the compiler not working
+            # (because it interprets the libs as input source files)
+            self._autotools.libs = list(tools.unix_path(l) for l in self._absolute_dep_libs_win) + self.deps_cpp_info.system_libs
```

Comment:
@uilianries: This idea is interesting for all autotools projects that have library dependencies.
I think it would be interesting to have this integrated in conan.

CC'ing @SSE4 @danimtb for their input.